src/main/java/seedu/trackbeau/logic/parser/FindCommandParser.java

```diff
@@ -1,33 +1,59 @@
 package seedu.trackbeau.logic.parser;
 
 import static seedu.trackbeau.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.trackbeau.commons.core.Messages.MESSAGE_UNKNOWN_COMMAND;
 
 import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 import seedu.trackbeau.logic.commands.FindCommand;
+import seedu.trackbeau.logic.commands.HelpCommand;
 import seedu.trackbeau.logic.parser.exceptions.ParseException;
-import seedu.trackbeau.model.customer.NameContainsKeywordsPredicate;
+import seedu.trackbeau.model.customer.SearchContainsKeywordsPredicate;
 
 /**
  * Parses input arguments and creates a new FindCommand object
  */
 public class FindCommandParser implements Parser<FindCommand> {
 
+    private static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<searchAreaKeyWord>\\S+)(?<arguments>.*)");
+
     /**
      * Parses the given {@code String} of arguments in the context of the FindCommand
      * and returns a FindCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
-    public FindCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
+    public FindCommand parse(String userInput) throws ParseException {
+        userInput = userInput.trim().replaceAll("\\n", "");
+        final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(userInput);
+        if (!matcher.matches()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
+        }
+
+        final String searchAreaKeyWord = matcher.group("searchAreaKeyWord");
+        final String arguments = matcher.group("arguments");
+
+        String trimmedArgs = arguments.trim();
         if (trimmedArgs.isEmpty()) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
-
-        return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        List<String> argumentList = Arrays.asList(trimmedArgs.split("\\s+"));
+
+        switch (searchAreaKeyWord.toLowerCase()) {
+        case "name":
+            return new FindCommand(new SearchContainsKeywordsPredicate("getName", argumentList));
+        case "phone":
+            return new FindCommand(new SearchContainsKeywordsPredicate("getPhone", argumentList));
```

Comment:
has the user guide been updated with the search commands?

Reply:
nope will be added on the next PR with additional searches